ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,211 @@
+    )
+    header.filters.add("LowQual", None, None, "Aggregated quality below threshold")
+    for field in fields_to_collect["numeric_aggregated_fields"]:
+        header.info.add(
+            field, orig_header.info[field].number, orig_header.info[field].type, orig_header.info[field].description
+        )
+    for field in fields_to_collect["numeric_array_fields"]:
+        header.info.add(field, ".", orig_header.info[field].type, orig_header.info[field].description)
+    for field in fields_to_collect["string_list_fields"]:
+        header.info.add(field, "1", orig_header.info[field].type, orig_header.info[field].description)
+    # exceptions: X_QUAL
+    header.info.add(FeatureMapFields.X_QUAL.value, ".", "Float", "Quality of reads containing this location")
+    header.formats.add(GT, 1, "String", "Genotype")
+    header.formats.add(DP, 1, "Integer", "Read depth")
+    header.formats.add(AD, "R", "Integer", "Allelic depths")
+    header.formats.add(VAF, 1, "Float", "Allele frequency")
+    # copy contigs from original header
+    for contig in orig_header.contigs:
+        header.contigs.add(contig)
+    header.add_sample(sample_name)
+
+    # open an output file
+    out_fh = pysam.VariantFile(output_vcf, "w", header=header)
+
+    cons_dict = defaultdict(dict)
+    with pysam.VariantFile(featuremap) as f:
+        prev_key = tuple()
+        for rec in f.fetch(chrom, start, end):
+            rec_id = (rec.chrom, rec.pos, rec.ref, rec.alts[0])
+            if FeatureMapFields.FILTERED_COUNT.value not in cons_dict[rec_id]:
+                if len(cons_dict.keys()) > 1:
+                    # write to file
+                    write_a_pileup_record(cons_dict[prev_key], prev_key, out_fh, header, min_qual)
+                    cons_dict.pop(prev_key)
+                # initialize the record
+                for field in fields_to_collect["numeric_aggregated_fields"]:
+                    cons_dict[rec_id][field] = 0
+                # exceptions: X_QUAL
+                cons_dict[rec_id]["X_QUAL"] = np.array([])
+                for field in fields_to_collect["numeric_array_fields"]:
+                    cons_dict[rec_id][field] = np.array([])
+                for field in fields_to_collect["string_list_fields"]:
+                    cons_dict[rec_id][field] = []
+            # update the record
+            for field in fields_to_collect["numeric_aggregated_fields"]:
+                cons_dict[rec_id][field] += 1
+            # exceptions: X_QUAL
+            cons_dict[rec_id]["X_QUAL"] = np.append(cons_dict[rec_id]["X_QUAL"], rec.qual)
+            for field in fields_to_collect["numeric_array_fields"]:
+                cons_dict[rec_id][field] = np.append(cons_dict[rec_id][field], rec.info[field])
+            for field in fields_to_collect["string_list_fields"]:
+                cons_dict[rec_id][field] += [rec.info[field]]
+            for field in fields_to_collect["fields_to_write_once"]:
```

Comment:
Do you want to make sure they are all identical? If they are not you will get the last one which seems arbitrary